### PR TITLE
Add variableHeight argument to CarouselProvider that disables the ratio

### PR DESCRIPTION
**Pull request by reckter**
**What**:

Add a `variableHeight` attribute to `CarouselProvider`.
If set to true, the Slider will not adhere to the ratio being set with `naturalSlideHeight` and `naturalSlideWidth`.
The Slider will have the height of the biggest Slide.

**Why**:

So you can use non scaling context (like text on the slide).
Without these changes, the Slide would be either too small or too big (and just clip the Slide in half).

This somewhat addresses the issues in the following issues.  This does not however change the hide of the Slider depending on the current shown slide, which might be another cool addition.

- https://github.com/express-labs/pure-react-carousel/issues/23
- https://github.com/express-labs/pure-react-carousel/issues/225
- https://github.com/express-labs/pure-react-carousel/issues/176
- https://github.com/express-labs/pure-react-carousel/issues/154

**How**:

We store `variableHeight` in the store in `CarouselProvider`.
Each `Slide` then sets styles depending on that.
The styles Reset the `position`, `height` and `padding-bottom` css attributes, which allows the Slider to be as heigh as the biggest Slide, therefore dynamicaly scalling with the content.

**Checklist**:

- [x] Documentation added/updated
- [x] Typescript definitions updated
- [x] Tests added and passing
- [x] Ready to be merged 

As this is my first change in the library, it might be wise to do a bit further testing, as I might have missed some egde cases.
But it feels very good, and all test pass.

**Workaround**
This can be archived "from the outside" by using these css rules:
```css
.slider li {
    position: unset !important;
    height: unset !important;
    padding-bottom: unset !important;
}
.slider li .carousel__inner-slide {
    position: unset;
}
```

### CODE REVIEW

**Comment by mrbinky3000** [comment]
Generally, if it can be handled via CSS, then I'm against adding new props to the components.  But this seems to be a recurring theme with the height so I relent.

I have some requests though...

Can you add an example to our list of live demos?  I want to download the branch and see the demo so I can see how this affects a vertical carousel?  

If one slide is 1000px in height, does that mean all slides are 1000px in height or are there "gaps" where the background will bleed through.  Have you looked at flex-box instead.  It will make all slides as tall as the tallest slide.

Is there an option for "variable width" on vertical carousels? 

I might want to change the name to useIntrinsicDimension or intrinsicAxisSize or something more descriptive because "variable height" to me sounds like responsive height, and the carousel is currently responsive as is.  Might be confusing to others.  I'm getting the Axis concept from flex-box.

**Comment by reckter** [comment]
@mrbinky3000 
I will add a Demo for sure.

Currently they do not flex to all become the same height.
Instead there is a gap beneath all smaller slides, that fills the missing height.
I am unsure what the right solution here would be regarding stretching all of the slides to be the same height, both (stretching and not-stretching) seem to make sense at first thought.

I haven't looked into the width case, so currently there is no such option, it shouldn't be too hard to add though. 

Open for renaming, haven't put too much thought into that.
`doNotEnforceRatio` might be good candidate as well, as this is essentially what this does.

**Comment by reckter** [comment]
Looking at the `vertical` orientation:

The width of the whole slider currently scales to 100% of the container, in both vertical and horizontal arrangement. Based on that width, all the other sizes are calculated.
But to have the same behavior in vertical as in horizontal mode (with the new flag enabled), the height should be the size set from outside and all the rest depending on it, so the width then can become dependent on the widest slide.

File: src/Slide/Slide.jsx [hunk 1]
```diff
@@ -116,6 +119,18 @@ const Slide = class Slide extends React.PureComponent {
...
+    if (intrinsicAxisSize) {
+      if (orientation === 'horizontal') {
+        tempStyle.height = 'unset';
+      } else {
+        tempStyle.width = 'unset';
+      }
+      tempStyle.position = 'unset';
+      tempStyle.paddingBottom = 'unset';
+      innerStyle.position = 'unset';
+    }
```
Comment (tim-steele) [inline src/Slide/Slide.jsx]: Question on this: instead of setting CSS via JavaScript, are we able to override the existing CSS with CSS?

**Comment by mrbinky3000** [comment]
Can you add the error for using this new flag in vertical mode?

Also can we change the name of the option to "isIntrinsicHeight".  We use "is[blank]" for booleans.  Also the word "use" implies a hook.

Make those changes and I will approve.

**Comment by reckter** [comment]
Let me know if the Error message is ok the way it is, or if you want to have something added :)

**Comment by mrbinky3000** [comment]
It's all good, thanks!

**Comment by janoist1** [comment]
Thanks for this!
When is it going to be released?

**Comment by larafale** [comment]
when release ? :) thanks for this package

**Comment by tim-steele** [comment]
@allcontributors please add @reckter for code

**Pull request merged**